src/Datadog.Trace/Plugins/IExtendPropagators.cs

```diff
@@ -0,0 +1,16 @@
+using Datadog.Trace.Propagation;
+
+namespace Datadog.Trace.Plugins
+{
+    /// <summary>
+    /// Provides propagators extendability point.
+    /// </summary>
+    public interface IExtendPropagators : IOTelExtension
```

Comment:
This interface looks very strange. It is like a provider for a provider. Can't we simply `IPropagatorsProvider` as the extendability point?

Reply:
The good thing is that if you use `IPropagatorsProvider` as the extensibility point then you can even dogfood it for the Datadog.Trace assembly and then we have the code already tested via integration test 😉

Reply:
removed that middle layer.